File: 2015/AI/AIClient_Python/src/event/UpdateMissileEvent.py
'''
Created on Jan 6, 2015

@author: scarriere
'''
from event.IngoingEvent import IngoingEvent
from aiclient.Singleton import Singleton
from world.World import World
from mathUtils.Vector2 import Vector2
from mathUtils.Direction import Direction

class UpdateMissileEvent(IngoingEvent):

    def __init__(self):
        self.teamId = 0
        self.characterId = 0
        self.position = Vector2(0,0)
        self.direction = Direction.UP
    
    def fillArguments(self, string):
        arguments = string.split(":", 5)
        
        self.teamId = int(arguments[0])
        self.characterId = int(arguments[1])
        self.position.x = int(arguments[2])
        self.position.y = int(arguments[3])
        self.direction = Direction(int(arguments[4]))
        
    def execute(self):
        world = Singleton(World)
        world.getTeam(self.teamId).characters[self.characterId].missile._updatePositionDirection(self.position, self.direction)
        

File: 2015/AI/AIClient_JAVA/src/event/UpdateMissileEvent.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** UpdateMissileEvent.java
 ** Event that happen when a missile move
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

import java.awt.Point;

import mathUtils.Direction;

import world.World;

public class UpdateMissileEvent extends IngoingEvent {
	private int teamID, characterID;
	private Direction direction;
	private Point position = new Point();

	@Override
	public void execute() {
		World.getInstance().getTeam(teamID).getCharacter(characterID)
				.getMissile().updateInfo(position, direction);
	}

	@Override
	public boolean fillArguments(String message) {
		String[] messageParts = message.split(SEPARATOR, 6);

		teamID = convertCharToNumeral(messageParts[0]);
		if (teamID == 0) {
			return false;
		}

		characterID = convertCharToNumeral(messageParts[1]);
		position.x = convertCharToNumeral(messageParts[2]);
		position.y = convertCharToNumeral(messageParts[3]);
		direction = Direction.values()[convertCharToNumeral(messageParts[4])];
		return true;
	}
}
